src/Symfony/Component/Routing/Loader/Psr4DirectoryLoader.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\FileLocatorInterface;
+use Symfony\Component\Config\Loader\FileLoader;
+use Symfony\Component\Routing\RouteCollection;
+
+final class Psr4DirectoryLoader extends FileLoader
+{
+    public function __construct(FileLocatorInterface $locator)
+    {
+        // PSR-4 directory loader has no env-aware logic, so we drop the $env constructor parameter.
+        parent::__construct($locator);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param array{path: string, psr4_prefix: string} $resource
```

Comment:
this type looks wrong to me

Reply:
Oh right, that was an old experiment but it didn't work out they way I wanted. 🙈